rules/utils/is-boolean.js

```diff
@@ -0,0 +1,205 @@
+'use strict';
+
+/**
+Determines whether a node is a Boolean Expression.
+
+@param {import('eslint').Rule.RuleContext} context
+@param {import('estree').Expression} node
+@param {number} [depth] - The current recursion depth. Users do not need to pass this parameter.
+@returns {boolean}
+*/
+function isBooleanExpression(context, node, depth = 0) {
```

Comment:
I know there is a method `isBooleanNode`, but it is not comprehensive enough, and it cannot track the type of Identifier.

https://github.com/sindresorhus/eslint-plugin-unicorn/blob/461b01cd69c42396ee1bda7ad583c4e2739b5696/rules/utils/boolean.js#L32-L37

So, I need more advice, should I use the same method.